# Position Controllers ---------------------------------------
joint1_velocity_controller:
  type:  effort_controllers/JointVelocityController
  joint: shoulder_pan_joint
  pid: {p: 2000.0, i: 100, d: 500.0}
joint2_velocity_controller:
  type:  effort_controllers/JointVelocityController
  joint: shoulder_lift_joint
  pid: {p: 2000.0, i: 100, d: 2000.0}
joint3_velocity_controller:
  type:  effort_controllers/JointVelocityController
  joint: elbow_joint
  pid: {p: 5000.0, i: 50, d: 1000.0}
joint4_velocity_controller:
  type:  effort_controllers/JointVelocityController
  joint: wrist_1_joint
  pid: {p: 2000.0, i: 50, d: 200.0}
joint5_velocity_controller:
  type:  effort_controllers/JointVelocityController
  joint: wrist_2_joint
  pid: {p: 2000.0, i: 50, d: 70.0}
joint6_velocity_controller:
  type:  effort_controllers/JointVelocityController
  joint: wrist_3_joint
  pid: {p: 2000.0, i: 50, d: 70.0}

## Publish all joint states -----------------------------------
#joint_state_controller:
#  type: joint_state_controller/JointStateController
#  publish_rate: 50


#ur_traj_controller:
#  type: effort_controllers/JointTrajectoryController
#  joints:
#    #- planar_joint_x   #required to fake move robot, as well definitions inside moveit_controller_configs, not required to plan
#    #- planar_joint_y   #required to fake move robot, as well definitions inside moveit_controller_configs, not required to plan
#    #- planar_joint_yaw #required to fake move robot, as well definitions inside moveit_controller_configs, not required to plan
#    - shoulder_pan_joint
#    - shoulder_lift_joint
#    - elbow_joint
#    - wrist_1_joint
#    - wrist_2_joint
#    - wrist_3_joint
#
#  constraints:
#    goal_time: 0.6
#    stopped_velocity_tolerance: 0.05
#    #planar_joint_x:   {trajectory: 0.1, goal: 0.05} #required to fake move robot, as well definitions inside moveit_controller_configs, not required to plan
#    #planar_joint_y:   {trajectory: 0.1, goal: 0.05} #required to fake move robot, as well definitions inside moveit_controller_configs, not required to plan
#    #planar_joint_yaw: {trajectory: 0.1, goal: 0.05} #required to fake move robot, as well definitions inside moveit_controller_configs, not required to plan
#    shoulder_pan_joint:   {trajectory: 0.1, goal: 0.1}
#    shoulder_lift_joint: {trajectory: 0.1, goal: 0.1}
#    elbow_joint: {trajectory: 0.1, goal: 0.1}
#    wrist_1_joint: {trajectory: 0.1, goal: 0.1}
#    wrist_2_joint: {trajectory: 0.1, goal: 0.1}
#    wrist_3_joint: {trajectory: 0.1, goal: 0.1}
#
#  gains:
#    #planar_joint_x:   {p:   100, d: 0.5, i: 0.01, i_clamp: 0} #required to fake move robot, as well definitions inside moveit_controller_configs, not required to plan
#    #planar_joint_y:   {p:   100, d: 0.5, i: 0.01, i_clamp: 0} #required to fake move robot, as well definitions inside moveit_controller_configs, not required to plan
#    #planar_joint_yaw: {p:   100, d: 0.5, i: 0.01, i_clamp: 0} #required to fake move robot, as well definitions inside moveit_controller_configs, not required to plan
#    shoulder_pan_joint:   {p: 100000, d: 0.5, i: 0.01, i_clamp: 0}
#    shoulder_lift_joint: {p: 7000, d: 1.0, i: 0.5, i_clamp: 0}
#    elbow_joint: {p: 5000, d: 1.0, i: 0.5, i_clamp: 0}
#    wrist_1_joint: {p:  1000, d: 0.1, i: 0.01, i_clamp: 0}
#    wrist_2_joint: {p:   100, d: 0.1, i: 0.01, i_clamp: 0}
#    wrist_3_joint: {p:   100, d: 0.1, i: 0.01, i_clamp: 0}
#
#  state_publish_rate:  50
#  action_monitor_rate: 10
#  stop_trajectory_duration: 0.5


# Trajectory controller
ur_traj_controller:
  type: "velocity_controllers/JointGroupVelocityController"
  joints:
    - shoulder_pan_joint
    - shoulder_lift_joint
    - elbow_joint
    - wrist_1_joint
    - wrist_2_joint
    - wrist_3_joint
  gains:
      #planar_joint_x:   {p:   100, d: 0.5, i: 0.01, i_clamp: 0} #required to fake move robot, as well definitions inside moveit_controller_configs, not required to plan
      #planar_joint_y:   {p:   100, d: 0.5, i: 0.01, i_clamp: 0} #required to fake move robot, as well definitions inside moveit_controller_configs, not required to plan
      #planar_joint_yaw: {p:   100, d: 0.5, i: 0.01, i_clamp: 0} #required to fake move robot, as well definitions inside moveit_controller_configs, not required to plan
    shoulder_pan_joint:   {p: 100000, d: 0.5, i: 0.01, i_clamp: 0}
    shoulder_lift_joint: {p: 7000, d: 1.0, i: 0.5, i_clamp: 0}
    elbow_joint: {p: 5000, d: 1.0, i: 0.5, i_clamp: 0}
    wrist_1_joint: {p:  1000, d: 0.1, i: 0.01, i_clamp: 0}
    wrist_2_joint: {p:   100, d: 0.1, i: 0.01, i_clamp: 0}
    wrist_3_joint: {p:   100, d: 0.1, i: 0.01, i_clamp: 0}
  constraints:
    goal_time: 0.6
    stopped_velocity_tolerance: 0.05
    shoulder_pan_joint: {trajectory: 0.1, goal: 0.1}
    shoulder_lift_joint: {trajectory: 0.1, goal: 0.1}
    elbow_joint: {trajectory: 0.1, goal: 0.1}
    wrist_1_joint: {trajectory: 0.1, goal: 0.1}
    wrist_2_joint: {trajectory: 0.1, goal: 0.1}
    wrist_3_joint: {trajectory: 0.1, goal: 0.1}
  stop_trajectory_duration: 0.5
  state_publish_rate: 25
  action_monitor_rate: 10